### DISCUSSION TITLE
Slow Oh-My-Zsh startup times on MacOS Sonoma 14.6.1

### DISCUSSION THREAD

**Discussion by hoseNdoW** [ohmyzsh/ohmyzsh#12642] [Help]
Hello,

I am experiencing slower than usual startup times when using oh-my-zsh on my MacBook Pro.

The issue is that my terminal initiation time has visibly increased since the update from the previous version. The time difference is quite noticeable. It was relatively fast before and now it seems to take a much longer time.

I tried commenting out parts of the config to find what might be causing such a load time increase, but it didn't help.

Even with a clean zsh, without any configs, and especially without oh-my-zsh, the load time has increased drastically for some reason, so I'm not sure that the problem is exactly here.

System specifications:
- Device: MacBook Pro (13-inch, 2020, Intel)
- OS: MacOS Sonoma 14.6.1 (23G93)

1. Benchmark results (with default zsh config)

```zsh
~/zsh-bench/zsh-bench
==> benchmarking login shell of user [username] ...
creates_tty=0
has_compsys=1
has_syntax_highlighting=0
has_autosuggestions=0
has_git_prompt=0
first_prompt_lag_ms=207.323
first_command_lag_ms=207.608
command_lag_ms=0.114
input_lag_ms=0.302
exit_time_ms=161.279
```

2. Benchmark results (with my zsh config)

```zsh
==> benchmarking login shell of user [username] ...
creates_tty=0
has_compsys=1
has_syntax_highlighting=0
has_autosuggestions=0
has_git_prompt=0
first_prompt_lag_ms=782.225
first_command_lag_ms=782.590
command_lag_ms=0.124
input_lag_ms=0.278
exit_time_ms=209.451
```

3. My zsh config file

```zsh
# Path to your oh-my-zsh installation.
export ZSH="$HOME/.oh-my-zsh"

plugins=(zsh-autosuggestions zsh-completions zsh-syntax-highlighting)

ZSH_THEME="robbyrussell"

source $ZSH/oh-my-zsh.sh

PROMPT="%B%F{magenta}%~%f %b"

eval "$(ssh-agent -s)" > /dev/null
ssh-add "$HOME/.ssh/[username]" > /dev/null

# fnm
eval "$(fnm env --use-on-cd)"
```

**Comment by mcornella** [comment]
Hi @hoseNdoW, seems like this is a general issue. If this is still reproducible with an empty `.zshrc` and `.zshenv` file, maybe something has changed in the [global files](https://zsh.sourceforge.io/Guide/zshguide02.html#l9). I'd check for all of those to see if there's anything that a command may have added that could be causing this.

You can also create a trace of the startup with `zsh -lxvic exit &>~/debug.log` and see everything that's executed on startup.
